src/BrightScriptDebugSession.ts

```diff
@@ -287,6 +306,157 @@ export class BrightScriptDebugSession extends DebugSession {
+                this.componentLibrariesStagingDirPaths.push(pathDetails);
+                await this.rokuDeploy.zipPackage(componentLibrary);
+            }
+            // #endregion
+
+            // #region prepare static file hosting
+            // maps file extension to MIME types
+            const mimeType = {
+                '.ico': 'image/x-icon',
+                '.html': 'text/html',
+                '.js': 'text/javascript',
+                '.json': 'application/json',
+                '.css': 'text/css',
+                '.png': 'image/png',
+                '.jpg': 'image/jpeg',
+                '.wav': 'audio/wav',
+                '.mp3': 'audio/mpeg',
+                '.svg': 'image/svg+xml',
+                '.pdf': 'application/pdf',
+                '.doc': 'application/msword',
+                '.eot': 'appliaction/vnd.ms-fontobject',
+                '.ttf': 'aplication/font-sfnt',
+                '.zip': 'application/zip'
+            };
+
+            http.createServer((req, res) => {
+                this.sendDebugLogLine(`${req.method} ${req.url}`);
+
+                // parse URL
+                const parsedUrl = url.parse(req.url);
+
+                // extract URL path
+                // Avoid https://en.wikipedia.org/wiki/Directory_traversal_attack
+                // e.g curl --path-as-is http://localhost:9000/../fileInDanger.txt
+                // by limiting the path to component libraries out directory only
+                const sanitizePath = path.normalize(parsedUrl.pathname).replace(/^(\.\.[\/\\])+/, '');
+                let pathname = path.join(this.componentLibrariesOutDir, sanitizePath);
+
+                fs.exists(pathname, function(exist) {
+                    if (!exist) {
+                        // if the file is not found, return 404
+                        res.statusCode = 404;
+                        res.end(`File ${pathname} not found!`);
+                        return;
+                    }
+
+                    // if is a directory
+                    // if (fs.statSync(pathname).isDirectory()) {
+                    // TODO: add support for directory listing
+                    // }
+
+                    // read file from file system
+                    fs.readFile(pathname, function(err, data) {
+                        if (err) {
+                            res.statusCode = 500;
+                            res.end(`Error getting the file: ${err}.`);
+                        } else {
+                            // based on the URL path, extract the file extension. e.g. .js, .doc, ...
+                            const ext = path.parse(pathname).ext;
+                            // if the file is found, set Content-type and send data
+                            res.setHeader('Content-type', mimeType[ext] || 'text/plain' );
+                            res.end(data);
+                        }
+                    });
+                });
+
+            }).listen(port);
```

Comment:
What happens if we restart the debug session (instead of stop and start). Does the http server get completely discarded and completely recreated? I'm just concerned that a restart will error out because the http server and port are already occupied.

Reply:
Have not had any problems with the server and or port being occupied after debugger stops or restarts.